Sources/Basics/Lock.swift

```diff
@@ -0,0 +1,35 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+
+/// A simple lock wrapper.
+public struct Lock {
```

Comment:
@Overcot wdyt about ditching the wrapper approach and instead just declare an extension on `NSLock` with the `withLock` function? then we can update the callosities to use `NSLock` instead of `LocK` from TSC

Reply:
Yeah, sure. If everyone aggrees, i would it. This seems like a better approach. Btw - is the NSLock available on Windows?

Reply:
> Btw - is the NSLock available on Windows?

I believe so